components/remote_settings/src/remote_settings.udl

```diff
@@ -73,3 +77,50 @@ interface RemoteSettings {
     [Throws=RemoteSettingsError]
     void download_attachment_to_path(string attachment_id, string path);
 };
+
+/// Stores application-level configuration data and constructs clients
+interface RemoteSettingsService {
+    constructor(RemoteSettingsConfig2 config);
+
+    /// Construct a new client with the configuration passed to the constructor
+    [Throws=RemoteSettingsError]
+    RemoteSettingsClient make_client(string collection_name);
+};
```

Comment:
This seems like a nice idea to me, but it's not totally clear that we need it.  How do others feel about this?

Reply:
I'm not quite sure I understand the discussion here. Would this fit with the following use case?

* iOS (or Android) code (swift/Kotlin) sets up the remote settings server to use stage + preview.
* Rust based search client calls `RemoteSettingsClient("search-config-v2").get()`.

Here I'd want the client to return the configuration according to the stage+preview server.

In other words, I would like the Rust based search client to be able to get the collection without having to worry about passing through the remote settings configuration.

I think we want the search client to be able to go directly to remote settings client as it'll avoid having to pass the configuration back and forth across the Uniffi interfaces.

Reply:
And further clarification, I think the Remote Settings server settings should be app-global rather than per-collection, hence I don't really want the search client to have to deal with those as well.

Reply:
Thanks for pointing this out.  My proposal was that the consumer app would create a `RemoteSettingsService` and use that as a factory for all the clients/collections.  `RemoteSettingsService` stores the server config and avoids the need for a global variable.

However, I can definitely see why that would be annoying for your use-case.  In order for you to create remote settings clients that used the correct server, the consumer app would need to pass the `RemoteSettingsService` to you.

A global variable could simplify things, but then you have to worry about startup order.  What happens if you constructed your client before the consumer app configured the server?  I'm not sure if this is a big deal or not, but I like the fact that the `RemoteSettingsService` makes this more explicit.

I don't have strong feelings either way, what do others think?